### PR TITLE
Kubemark scripts retry failed gcloud commands

### LINKED ISSUE #18334: Kubemark - add retries in scripts.

### PR DESCRIPTION
**Pull request by gmarek**
Fixes #18334
cc @ihmccreery

### CODE REVIEW

File: test/kubemark/common.sh [hunk 1]
```diff
@@ -23,3 +24,27 @@ export PROJECT
 MASTER_NAME="${INSTANCE_PREFIX}-kubemark-master"
 MASTER_TAG="kubemark-master"
 EVENT_STORE_NAME="${INSTANCE_PREFIX}-event-store"
+
+RETRIES=3
+
+# arguments:
+# $1: name of operation to embed in an error message
+# $2...: all stuff that goes after 'gcloud compute '
+function run-gcloud-compute {
```
Comment (bprashanth) [inline test/kubemark/common.sh]: do you want to move this to some place more usable from other setup scripts? (I assume they all have similar retry problems)

**Comment by bprashanth** [comment]
LGTM but my bash experience is limited to scripts for local testing, so reassigning to Isaac

File: test/kubemark/common.sh (hunk 1 above)
Comment (gmarek) [inline test/kubemark/common.sh]: Not sure if it's worth it. This code is already there, though it's duplicated all over the place. We can leave this as a cleanup for someone.

File: test/kubemark/common.sh [hunk 2]
```diff
@@ -23,3 +24,27 @@ export PROJECT
 MASTER_NAME="${INSTANCE_PREFIX}-kubemark-master"
 MASTER_TAG="kubemark-master"
 EVENT_STORE_NAME="${INSTANCE_PREFIX}-event-store"
+
+RETRIES=3
+
+# arguments:
+# $1: name of operation to embed in an error message
```
Comment (ihmccreery) [inline test/kubemark/common.sh]: This seems unnecessary; why not just use `$2`?  It's extra flags in error messages, but those are often important to know about.

File: test/kubemark/common.sh [hunk 3]
```diff
@@ -23,3 +24,27 @@ export PROJECT
...
+# $2...: all stuff that goes after 'gcloud compute '
```
Comment (ihmccreery) [inline test/kubemark/common.sh]: Make this comment $@, per above removal of $1?

File: test/kubemark/common.sh [hunk 4]
```diff
@@ -23,3 +24,27 @@ export PROJECT
...
+function run-gcloud-compute {
+  local attempt=0
+  local name=$1
+  shift
+  while true; do
```
Comment (ihmccreery) [inline test/kubemark/common.sh]: Consider using something like

```
for retry in $(seq 1 3); do
```

See examples in `cluster/gce/util.sh`.

File: test/kubemark/common.sh (hunk 1 above)
Comment (ihmccreery) [inline test/kubemark/common.sh]: I agree with @bprashanth; please file an issue to clean up.

File: test/kubemark/common.sh (hunk 2 above)
Comment (gmarek) [inline test/kubemark/common.sh]: Because we might want to use it with `gcloud compute ssh`, but I can leave this change for someone who'll want to do it.

**Comment by gmarek** [comment]
@ihmccreery PTAL

**Comment by davidopp** [comment]
@ihmccreery says "needs more commentary, and it's removing quotes which is against the bash style guide."

@fgrzadkowski someone needs to take over this PR ASAP and address Ike's comments.

**Comment by fgrzadkowski** [comment]
@mwielgus FYI

**Pull request closed without merging**